frontend/src/metabase/nav/components/SearchResults.tsx

```diff
@@ -76,7 +75,7 @@ export const SearchResults = ({
 
   const hasResults = list.length > 0;
 
-  if (isLoading) {
+  if (isLoading || isWaitingForDebounce) {
```

Comment:
Keeps the loading spinner around when the user is typing, so we're not showing old results when the user is typing